Microsoft.Toolkit.Uwp.UI.Controls/GridSplitter/GridSplitter.cs

```diff
@@ -0,0 +1,170 @@
+
+                if ((gridSplitterSiblingColumnIndex >= 0)
+                    && (gridSplitterSiblingColumnIndex < Resizable.ColumnDefinitions.Count))
+                {
+                    return Resizable.ColumnDefinitions[gridSplitterSiblingColumnIndex];
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the current Row definition of the parent Grid
+        /// </summary>
+        private RowDefinition CurrentRow
+        {
+            get
+            {
+                if (Resizable == null)
+                {
+                    return null;
+                }
+
+                var gridSplitterTargetedRowIndex = GetTargetedRow();
+
+                if ((gridSplitterTargetedRowIndex >= 0)
+                    && (gridSplitterTargetedRowIndex < Resizable.RowDefinitions.Count))
+                {
+                    return Resizable.RowDefinitions[gridSplitterTargetedRowIndex];
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the Sibling Column definition of the parent Grid
+        /// </summary>
+        private RowDefinition SiblingRow
+        {
+            get
+            {
+                if (Resizable == null)
+                {
+                    return null;
+                }
+
+                var gridSplitterSiblingRowIndex = GetSiblingRow();
+
+                if ((gridSplitterSiblingRowIndex >= 0)
+                    && (gridSplitterSiblingRowIndex < Resizable.ColumnDefinitions.Count))
+                {
+                    return Resizable.RowDefinitions[gridSplitterSiblingRowIndex];
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GridSplitter"/> class.
+        /// </summary>
+        public GridSplitter()
+        {
+            this.DefaultStyleKey = typeof(GridSplitter);
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            if (_splitter != null)
+            {
+                // Unhook registered events
+                _splitter.Loaded -= GridSplitter_Loaded;
+                _splitter.DragStarted -= Splitter_DragStarted;
+                _splitter.DragDelta -= Splitter_DragDelta;
+                _splitter.DragCompleted -= Splitter_DragCompleted;
+            }
+
+            _splitter = GetTemplateChild(SplitterName) as Thumb;
+            _iconDisplay = GetTemplateChild(IconDisplayName) as TextBlock;
+            if (_splitter == null)
+            {
+                return;
+            }
+
+            // Register Events
+            _splitter.Loaded += GridSplitter_Loaded;
+            _splitter.DragStarted += Splitter_DragStarted;
+            _splitter.DragDelta += Splitter_DragDelta;
+            _splitter.DragCompleted += Splitter_DragCompleted;
```

Comment:
Can't these just be on this instead? (You can use Manipulation* events instead of Drag* events)